scripts/generators/typescript.js

```diff
@@ -0,0 +1,186 @@
+"use strict";
+
+const t = require("../../packages/babel-types");
+const utils = require("./utils");
+
+let code = `// NOTE: This file is autogenerated. Do not modify.
+// See scripts/generators/typescript.js for script used.
+
+export interface Comment {
+  value: string;
+  start: number;
+  end: number;
+  loc: SourceLocation;
+  type: "BlockComment" | "LineComment";
```

Comment:
This type prevents refinements (if you have something of type Comment, you can't compare its `type` to refine the type).

It's not harmful in practice I guess since both BlockComment and LineComment don't actually add anything to the Comment type.

Reply:
Good catch - updated.